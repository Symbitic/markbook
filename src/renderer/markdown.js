import frontmatter from 'remark-frontmatter'
import html from 'rehype-stringify'
import include from 'remark-include'
import katex from 'rehype-katex'
import math from 'remark-math'
import markdown from 'remark-parse'
import remark2rehype from 'remark-rehype'
import unified from 'unified'
import yamlConfig from 'remark-yaml-config'
import rename from './remark-rename'
import meta from './remark-meta'
import deflist from './remark-deflist'
import supersub from './remark-supersub'

const createProcessor = () =>
  unified()
    .use(markdown, {
      footnotes: true
    })
    .use(frontmatter)
    .use(yamlConfig)
    .use(include)
    .use(math)
    .use(deflist)
    .use(supersub)
    .use(meta)
    .use(rename)
    .use(remark2rehype)
    .use(katex)
    .use(html)

export default createProcessor
