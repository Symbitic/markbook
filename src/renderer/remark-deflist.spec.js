/* global expect */
import deflist from './remark-deflist'
import html from 'rehype-stringify'
import markdown from 'remark-parse'
import remark2rehype from 'remark-rehype'
import unified from 'unified'

const strip = ([str]) => str.replace(/\n +/g, '\n')

const parse = str =>
  unified()
    .use(markdown)
    .use(deflist)
    .use(remark2rehype)
    .use(html)
    .process(str)
    .then(data => data.toString())

const tests = [
  [
    'basic definition list',
    strip`
      Term 1
      
      : Definition 1
    `
  ],
  [
    'basic definition list + bold',
    strip`
      Term 1

      : Definition **1**
    `
  ]
]

describe.each(tests)('remark-deflist', (name, source) => {
  it(`should parse a ${name}`, () => {
    expect(parse(source)).resolves.toMatchSnapshot()
  })
})
