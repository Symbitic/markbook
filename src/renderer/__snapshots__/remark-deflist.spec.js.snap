// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-deflist should parse a basic definition list + bold 1`] = `"<p><dl><dt>Term 1</dt><dd>Definition <strong>1</strong></dd></dl></p>"`;

exports[`remark-deflist should parse a basic definition list 1`] = `"<p><dl><dt>Term 1</dt><dd>Definition 1</dd></dl></p>"`;
